### DISCUSSION TITLE
What is a good value for nSamples in EstimateBetweenness?

### DISCUSSION THREAD

**Discussion by Ishitori** [networkit/networkit#989] [Q&A] [answered]
Hi there,

I managed to run in parallel EstimateBetweenness algorithm on my graph of 350M nodes and 5.5B edges with only nSamples=5. Any value significantly larger gives me OOM even on my huge EC2 instance.

To fight OOM, but still run in parallel, I came up with an idea of limiting number of OpenMP threads by setting OMP_THREAD_LIMIT. Using that approach, I could increase nSamples to 50 and most probably can go higher.

Now, the question are there any rules of thumbs of how to estimate the "appropriate" value of nSamples given my graph size? I have read the paper, but I am not sure what nSamples in the code stands for (is it "k" in the paper)? Any guidance would be appreciated.

**Comment by rionda** [comment, marked as answer]
Not sure what paper you are referring to, but in the [original paper](http://rionda.to/papers/RiondatoKornaropoulos-BetweennessSampling-DMKD.pdf), we use `r` for the sample size (see, e.g., equation (3)). I don't think 50 is sufficiently large: a reasonable sample size would be at least in the thousands.
